jobs/resource_job.go

```diff
@@ -74,6 +74,17 @@ type TaskDependency struct {
 	TaskKey string `json:"task_key,omitempty"`
 }
 
+// BEGIN Jobs + Repo integration preview
+type GitSource struct {
+	GitUrl      string `json:"git_url"`
+	GitProvider string `json:"git_provider"`
```

Comment:
Try to guess provider, like this: https://github.com/databrickslabs/terraform-provider-databricks/blob/master/workspace/resource_repo.go#L49 - reuse `getProviderFromUrl` function (it needs to be exported)